app/Models/File.php

```diff
@@ -153,16 +153,22 @@ public function getInfo()
      * Sync the song with all available media info against the database.
      *
      * @param array $tags  The (selective) tags to sync (if the song exists)
-     * @param bool  $force Whether to force syncing, even if the file is unchaged
+     * @param bool  $force Whether to force syncing, even if the file is unchanged
+     * @param bool  $path  Whether to force syncing of a specific path, even
+     *                     if the file is unchanged
      *
      * @return bool|Song A Song object on success,
      *                   true if file exists but is unmodified,
      *                   or false on an error.
      */
-    public function sync($tags, $force = false)
+    public function sync($tags, $force = false, $path = null)
     {
         // If the file is not new or changed and we're not forcing update, don't do anything.
-        if (!$this->isNewOrChanged() && !$force) {
+        if (!$this->isNewOrChanged() && !$force && !$path) {
+            return true;
+        }
+
+        if ($path && !preg_match('/'.preg_quote($path, '/').'/', $this->getPath())) {
```

Comment:
@phanan Sorry, missed this. It shouldn't matter on windows since we're not doing any system checks for an existing directory or location, we're simply checking if the `path` of the File model object matches regular expression of the string specified by the user.

If the user uses backslashes because it is a Windows system, the `preg_match` will still work the same way as long as the `File` model also returns backslashes for its `path` property.